### PR TITLE
chore(animations): import AST and CompileMetadata with the `*` alias

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

File: modules/@angular/core/src/animation/animation_style_util.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {StringMapWrapper} from '../facade/collection';
 import {isPresent} from '../facade/lang';
 
 import {FILL_STYLE_FLAG} from './animation_constants';
-import {AUTO_STYLE} from './metadata';
```
Comment (vicb) [inline modules/@angular/core/src/animation/animation_style_util.ts]: can't we do just the opposite and drop the `animation_` prefix (we are in the `animation` folder already)

Comment (matsko) [inline modules/@angular/core/src/animation/animation_style_util.ts]: when there is a prefix.

**Comment by vicb** [comment]
status ?

File: modules/@angular/compiler/src/animation/animation_compiler.ts [hunk 2]
```diff
@@ -12,14 +12,14 @@ import {Identifiers, createIdentifier} from '../identifiers';
 import * as o from '../output/output_ast';
 import {ANY_STATE, DEFAULT_STATE, EMPTY_STATE} from '../private_import_core';
 
-import {AnimationAst, AnimationAstVisitor, AnimationEntryAst, AnimationGroupAst, AnimationKeyframeAst, AnimationSequenceAst, AnimationStateDeclarationAst, AnimationStateTransitionAst, AnimationStepAst, AnimationStylesAst} from './animation_ast';
+import * as asts from './animation_ast';
```
Comment (mhevery) [inline modules/@angular/compiler/src/animation/animation_compiler.ts]: I am not a fan of this syntax. What is the reasoning? We don't do it anywhere else.

**Review by mhevery — CHANGES_REQUESTED** [review]
Other changes are fine. I am not a fan of `import * as foo` Can we revert that? What is your reasoning?

File: modules/@angular/compiler/src/animation/animation_compiler.ts (hunk 2 above)
Comment (matsko) [inline modules/@angular/compiler/src/animation/animation_compiler.ts]: All of the items in here are imported and used within the parser/compiler code. It cleans down the code a lot on the import level. We do this all over the place with the codegen `output_ast.ts` code (https://github.com/angular/angular/blob/master/modules/@angular/compiler/src/compiler_util/render_util.ts#L12) and in the metadata_resolver with the `compile_metadata.ts` classes (https://github.com/angular/angular/blob/master/modules/@angular/compiler/src/metadata_resolver.ts#L14)

Comment (mhevery) [inline modules/@angular/compiler/src/animation/animation_compiler.ts]: The counterargument is that it is not clear which of the symbols are actually needed from the import. 

I think we should be consistent in the way we do imports, and most place it have it the other way. We should probably discuss this in our next meeting.

**Pull request closed without merging**